internal/translayer/auditing/translate.go

```diff
@@ -0,0 +1,56 @@
+package auditing
+
+import (
+	"fmt"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/pointer"
+)
+
+type AuditingConfigType string
+
+const (
+	None          AuditingConfigType = "NONE"
+	FilterBuilder AuditingConfigType = "FILTER_BUILDER"
+	FilterJSON    AuditingConfigType = "FILTER_JSON"
+)
+
+type Auditing struct {
+	Enabled                   bool
+	AuditAuthorizationSuccess bool
+	ConfigurationType         AuditingConfigType
+	AuditFilter               string
```

Comment:
In the CRD is possible to have a formatted representation of the JSON, but I think we should check if Atlas returns this in the same format we set, otherwise, we need normalization.
we would fail comparing `{"a":"b"}` with
```
{
  "a": "b"
}
```

Reply:
Right! problem is, the Spec is added in the next PR. So not sure if this is the right place to fix, as the type changes quite a lot in this case.

Reply:
Definitely this is fixed in the next PR #1594, we use this:

```go
	// AuditFilter contains the JSON/YAML definition of the audit logging filter
	AuditFilter *apiextensionsv1.JSON `json:"auditFilter,omitempty"`
```